### PR TITLE
Haddad surah layout impl

### PR DESCRIPTION
**Pull request by IamHaddad**

### CODE REVIEW

File: surah/src/main/res/layout/row_aya.xml [hunk 1]
```diff
@@ -1,32 +1,61 @@
...
+        <TextView
+            android:id="@+id/tvAyaNumber"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_centerInParent="true"
+            android:textAlignment="center"
+            android:textColor="@android:color/white"
+            tools:text="286" />
+
+    </RelativeLayout>
```
Comment (helloseyedjafari) [inline surah/src/main/res/layout/row_aya.xml]: you can replace this with:
```
<TextView
            android:id="@+id/tvAyaNumber"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:textAlignment="center"
            android:gravity="center"
            android:textColor="@android:color/white"
            android:background="@drawable/ic_number_circle"
            tools:text="286" />
```

File: surah/src/main/res/layout/row_aya.xml [hunk 2]
```diff
@@ -1,32 +1,61 @@
...
```
Comment (helloseyedjafari) [inline surah/src/main/res/layout/row_aya.xml]: I think we need a linearLayout to hold the toolbar for each aya

File: surah/src/main/res/drawable/ic_bismillah.xml [hunk 3]
```diff
@@ -0,0 +1,93 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (helloseyedjafari) [inline surah/src/main/res/drawable/ic_bismillah.xml]: I think it's better to use a png or use a font on the text

File: surah/src/main/res/layout/view_surah.xml [hunk 4]
```diff
@@ -7,37 +7,131 @@
...
+        android:layout_height="wrap_content"
+        android:text="Alrahman"
+        android:textAlignment="center"
+        android:textColor="@color/black"
+        app:layout_constraintBottom_toTopOf="@+id/cardViewMainSurah"
         app:layout_constraintLeft_toLeftOf="parent"
         app:layout_constraintRight_toRightOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
 
+    <RelativeLayout
```
Comment (helloseyedjafari) [inline surah/src/main/res/layout/view_surah.xml]: I think we need a CardView here

**Review by helloseyedjafari — COMMENTED** [review]
please consider my comments

File: surah/src/main/res/layout/row_aya.xml (hunk 2 above)
Comment (IamHaddad) [reply surah/src/main/res/layout/row_aya.xml]: Toolbar? I thought it was removed in the last design

File: surah/src/main/res/drawable/ic_bismillah.xml (hunk 3 above)
Comment (IamHaddad) [reply surah/src/main/res/drawable/ic_bismillah.xml]: PNGs are not scalable

File: surah/src/main/res/drawable/ic_number_circle.xml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  <item>
+    <shape android:shape="oval">
+      <gradient
+          android:centerColor="#ffe2f1f6"
+          android:endColor="#ffe2f1f6"
+          android:gradientRadius="56dp"
+          android:startColor="#802BB5A8"
+          android:type="radial" />
+      <padding
+          android:bottom="6dp"
```
Comment (rezaiyan) [inline surah/src/main/res/drawable/ic_number_circle.xml]: Less padding looks the circle like clipped, 36dp is more pretty

Comment (IamHaddad) [reply surah/src/main/res/drawable/ic_number_circle.xml]: I just looked and didn't see any shadow in Figma design sample, I'm going to remove it anyway

File: surah/src/androidMain/kotlin/com/islamversity/surah/view/SurahHeader.kt [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        binding.tvSurahNameTranslate.text = surahHeaderUIModel.nameTranslate
+        binding.tvSurahOrigin.text = surahHeaderUIModel.origin
+        binding.tvSurahVerses.text = surahHeaderUIModel.verses.toString()
+
+        binding.tvSurahNumber.textSize = surahHeaderUIModel.fontSize.toFloat()
+        binding.tvSurahName.textSize = surahHeaderUIModel.fontSize.toFloat()
+        binding.tvSurahNameTranslate.textSize = surahHeaderUIModel.fontSize.toFloat()
+        binding.tvSurahOrigin.textSize = surahHeaderUIModel.fontSize.toFloat()
+        binding.tvSurahVerses.textSize = surahHeaderUIModel.fontSize.toFloat()
+
```
Comment (alirezanazari) [inline surah/src/androidMain/kotlin/com/islamversity/surah/view/SurahHeader.kt:31]: dear @neptu39701 
i think it's better to use `Kotlin Scope Functions` like `apply{...}` here.
there are lots of `binding`s and they could be removed and cods be more readable.

also `surahHeaderUIModel `
one of them could be remove.

File: surah/src/main/res/drawable/aya_circle_shape.xml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+                android:centerColor="#ffe2f1f6"
+                android:endColor="#ffe2f1f6"
+                android:gradientRadius="56dp"
+                android:startColor="#802BB5A8"
+                android:type="radial" />
+            <padding
+                android:bottom="36dp"
+                android:left="36dp"
+                android:right="36dp"
+                android:top="36dp" />
```
Comment (alirezanazari) [inline surah/src/main/res/drawable/aya_circle_shape.xml]: dimensions and colors could be referenced from resources(dimens/colors) and make future changes easier.

File: surah/src/androidMain/kotlin/com/islamversity/surah/view/SurahHeader.kt (hunk 6 above)
Comment (IamHaddad) [reply surah/src/androidMain/kotlin/com/islamversity/surah/view/SurahHeader.kt:31]: Dear @alirezanazari thank you for your review, please consider the first part (bind using apply) is done!
Can you explain the second part ? What is with the surahHeaderUIModel ?

File: surah/src/main/res/drawable/aya_circle_shape.xml (hunk 7 above)
Comment (IamHaddad) [reply surah/src/main/res/drawable/aya_circle_shape.xml]: Sorry this file was not supposed to be here anymore, thanks for reminding

File: surah/src/androidMain/kotlin/com/islamversity/surah/view/SurahHeader.kt (hunk 6 above)
Comment (alirezanazari) [reply surah/src/androidMain/kotlin/com/islamversity/surah/view/SurahHeader.kt:31]: thank you @neptu39701 
i mean these lines could be shorter and more readable:
```
        binding.apply {
            tvSurahNumber.text = surahHeaderUIModel.number
            tvSurahName.text = surahHeaderUIModel.name
            .....
        }
        
```
        or
```
        surahHeaderUIModel.apply {
            binding.tvSurahNumber.text = number
            binding.tvSurahName.text = name
            .....
        }

```

Comment (IamHaddad) [reply surah/src/androidMain/kotlin/com/islamversity/surah/view/SurahHeader.kt:31]: thanks i went with the first one

**Pull request merged**